src/main/java/com/gregtechceu/gtceu/common/data/GTItems.java

```diff
@@ -476,6 +423,27 @@ public Component getItemName(ItemStack stack) {
             .onRegister(materialInfo(new ItemMaterialInfo(new MaterialStack(GTMaterials.Glass, GTValues.M / 4))))
             .register();
 
+    public static ItemEntry<ComponentItem> createFluidCell(Material mat, int capacity, int matSize, int stackSize) {
+        var prop = mat.getProperty(PropertyKey.FLUID_PIPE);
+        if (prop == null) {
+            GTCEu.LOGGER.error("Material {} does not have any fluid props!", mat.getName());
+            throw new IllegalArgumentException();
+        }
```

Comment:
Why would you log and then throw? Just throw with a message so the log didn't get eaten up by the stacktrace.
Or, just use `Preconditions`.

```suggestion
        var prop = mat.getProperty(PropertyKey.FLUID_PIPE);
        Preconditions.checkArgument(prop != null, "Material { %s } does not have Fluid Pipe properties, but is being used to create a Fluid Cell", mat.getName());
```

Or some variation of the sort.